### PR TITLE
samples/susbys/mgmt/mcumgr/smp_svr: add nrf5 configs

### LINKED ISSUE #24442: samples/subsys/mgmt/mcumgr/smp_svr: should enable BT and FS for nrf52 boards
**Describe the bug**
The sample used to build with BT transport and FS command set enabled for out-of the box build. After merge https://github.com/zephyrproject-rtos/zephyr/pull/20983 its stops.

**To Reproduce**
1.` west build -d build/smp_svr -b nrf52840dk_nrf52840 ./zephyr/samples/subsys/mgmt/mcumgr/smp_svr/`

**Expected behavior**
These feature should be enabled by default for nrf52 targets

**Impact**
bad user experience

**Environment (please complete the following information):**
 - runing master https://github.com/zephyrproject-rtos/zephyr/commit/762b5219310e5e8acca0d02bab4c3f883f5a4d81

### PR DESCRIPTION
**Pull request by nvlsianpu**
Added default configuration files for nrf5 boards so the sample
is building with expected features for these boards.

Would be greater to use OVERLAY_CONFIG instead of files addition,
but that doesn't work from CMakeLists.txt scope.

fixes #24442

### CODE REVIEW

**Comment by nvlsianpu** [comment]
Would be greater to use OVERLAY_CONFIG instead of files addition,
but that doesn't work from CMakeLists.txt scope.

```cmake
if((BOARD STREQUAL nrf52840dk_nrf52840) OR
   (BOARD STREQUAL nrf52dk_nrf52832))
  set(OVERLAY_CONFIG overlay-bt.conf)
endif()
```
@ioannisg any hint on above doubt?

**Comment by aunsbjerg** [comment]
Maybe I'm in the wrong here, but I don't see why adding separate board configs is necessary when they essentially copy the already existing `overlay-bt.conf` and `overlay-bt-tiny.conf`. To use an overlay only requires adding `-- -DOVERLAY_CONFIG=overlay-bt.conf` to `west build`.

File: subsys/mgmt/Kconfig [hunk 1]
```diff
@@ -11,6 +11,7 @@ config MCUMGR_SMP_BT
 	select BT
 	select BT_PERIPHERAL
 	select BT_GATT_DYNAMIC_DB
+	select LEGACY_TIMEOUT_API
```
Comment (aunsbjerg) [inline subsys/mgmt/Kconfig:14]: This is already selected in the `BT` kconfig file here `zephyr/subsys/bluetooth/Kconfig`

Comment (nvlsianpu) [reply subsys/mgmt/Kconfig:14]: right, oversaw

**Comment by de-nordic** [comment]
When i try to build the nrf51dk_nrf51422:
```
 west build -b nrf51dk_nrf51422  --build-dir smp_svr zephyr/samples/subsys/mgmt/mcumgr/smp_svr -- -DOVERLAY_CONFIG=boards/nrf51dk_nrf51422.conf
```
I get this:
```
zephyr-sdk-0.11.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: region `FLASH' overflowed by 14708 bytes
````
Can someone tell me how to build this correctly?

**Comment by nvlsianpu** [comment]
^^ need to be build without eg. looging for meet slot constrains

File: subsys/mgmt/Kconfig (hunk 1 above)
Comment (nvlsianpu) [reply subsys/mgmt/Kconfig]: thanks to https://github.com/zephyrproject-rtos/zephyr/pull/24341 mcumgr library supports new timer API now.

**Comment by nvlsianpu** [comment]
@de-nordic I disable logging so nf52dk builds works now,

File: samples/subsys/mgmt/mcumgr/smp_svr/boards/nrf52840dk_nrf52840 copy.conf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Allow for large Bluetooth data packets.
```
Comment (de-nordic) [inline samples/subsys/mgmt/mcumgr/smp_svr/boards/nrf52840dk_nrf52840 copy.conf:1]: This file seems to be weird named copy of samples/subsys/mgmt/mcumgr/smp_svr/boards/nrf52840dk_nrf52840.conf,  is that ok?

**Review by de-nordic — COMMENTED** [review]
> 
> 
> @de-nordic I disable logging so nf52dk builds works now,

Thanks, now it builds.

File: samples/subsys/mgmt/mcumgr/smp_svr/boards/nrf52840dk_nrf52840 copy.conf (hunk 2 above)
Comment (nvlsianpu) [reply samples/subsys/mgmt/mcumgr/smp_svr/boards/nrf52840dk_nrf52840 copy.conf]: thx - this intended to be nrf52dk one.

**Review by de-nordic — APPROVED** [review]
Seems fine.

**Comment by aunsbjerg** [comment]
If this is the approach this sample is taking, do we need to keep `overlay-bt.conf` and `overlay-bt-tiny.conf`? And what about `overlay-udp.conf`, do we keep that and not add board specific configs for UDP samples? 

In any case, the sample documentation should be updated to reflect the changes made here.

**Comment by nvlsianpu** [comment]
@aunsbjerg I would keep these overaly as might be used for any boards.
Per-board config I added are cure for out-of-the box experience.

I can say that would be great if overlay-config might work for these board instead, but it does work only in sanity-check framework. I will open future request for that - anyway it is separate issue.

**Comment by aunsbjerg** [comment]
@nvlsianpu What do you mean that overlay config doesn't work for these boards? Using overlay configs with `west build` works fine for the freedom board and I can at least build for the nordic boards as well.

I can understand wanting to improve the out of box experience, but I don't understand how this is improving it much? You still have to specify a board specific overlay using `-DOVERLAY_CONFIG=boards/<board>.conf`, so what's the difference from just using the more generic overlay configs already existing? It's just more stuff to maintain later.

Anyway, it's not my decision, I just don't see much value being added here, only added complexity.

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
> I can understand wanting to improve the out of box experience, but I don't understand how this is improving it much? You still have to specify a board specific overlay using `-DOVERLAY_CONFIG=boards/<board>.conf`, so what's the difference from just using the more generic overlay configs already existing? It's just more stuff to maintain later.

No, overlays in the `boards/` folder are picked up automagically by the build system if they are called `boards/<board>.conf` or `boards/<board>.overlay`.

**Pull request merged**